bouncer/tests/broker_level_screening.ts

```diff
@@ -0,0 +1,194 @@
+import { randomBytes } from 'crypto';
+import { execSync } from 'child_process';
+import { ExecutableTest } from '../shared/executable_test';
+import { sendBtcAndReturnTxId } from '../shared/send_btc';
+import {
+  hexStringToBytesArray,
+  newAddress,
+  sleep,
+  handleSubstrateError,
+  brokerMutex,
+} from '../shared/utils';
+import { getChainflipApi, observeEvent } from '../shared/utils/substrate';
+import Keyring from '../polkadot/keyring';
+import { requestNewSwap } from '../shared/perform_swap';
+import { FillOrKillParamsX128 } from '../shared/new_swap';
+import { getBtcBalance } from '../shared/get_btc_balance';
+
+const keyring = new Keyring({ type: 'sr25519' });
+const broker = keyring.createFromUri('//BROKER_1');
+
+/* eslint-disable @typescript-eslint/no-use-before-define */
+export const testBrokerLevelScreening = new ExecutableTest('Broker-Level-Screening', main, 300);
+
+/**
+ * Observes the balance of a BTC address and returns true if the balance changes. Times out after 100 seconds and returns false if the balance does not change.
+ *
+ * @param address - The address to observe the balance of.
+ * @returns - Whether the balance changed.
+ */
+async function observeBtcAddressBalanceChange(address: string): Promise<boolean> {
+  const MAX_RETRIES = 100;
+  const initialBalance = await getBtcBalance(address);
+  for (let i = 0; i < MAX_RETRIES; i++) {
+    await sleep(1000);
+    const balance = await getBtcBalance(address);
+    if (balance !== initialBalance) {
+      return Promise.resolve(true);
+    }
+  }
+  console.error(`BTC balance for ${address} did not change after ${MAX_RETRIES} seconds.`);
+  return Promise.resolve(false);
+}
+
+/**
+ * Submits a transaction as tainted to the extrinsic on the state chain.
+ *
+ * @param txId - The txId to submit as tainted as byte array in the correct order.
```

Comment:
```suggestion
 * @param txId - The txId to submit as tainted as byte array in the order it is on the Bitcoin chain - which is reverse of how it's normally displayed in block explorers.
```
correct is a bit ambiguous